app/ts/components/Transactions.tsx

```diff
@@ -1,65 +1,129 @@
-	blockInfo: Signal<BlockInfo>
-	signers: Signal<Signers>
-	appSettings: Signal<AppSettings>
-	fundingAmountMin: ReadonlySignal<bigint>
+  interceptorPayload: Signal<BundleState | undefined>
+  blockInfo: Signal<BlockInfo>
+  signers: Signal<Signers>
+  appSettings: Signal<AppSettings>
+  fundingAmountMin: ReadonlySignal<bigint>
 }) => {
-	const transactions = createBundleTransactions(
-		interceptorPayload.peek(),
-		signers.peek(),
-		blockInfo.peek(),
-		appSettings.peek().blocksInFuture,
-		fundingAmountMin.peek(),
-	)
+  const transactions = computed(() =>
+    createBundleTransactions(interceptorPayload.peek(), signers.peek(), blockInfo.peek(), appSettings.peek().blocksInFuture, fundingAmountMin.peek()),
+  )
+
+  const parsedTransactions = useSignal<(FlashbotsBundleTransaction & { decoded?: JSXInternal.Element })[]>(transactions.peek())
+  const parseTransactionsCb = async () => {
+    try {
+      const uniqueAddresses = [...new Set(transactions.value.map((x) => x.transaction.to))]
+      // @TODO: Map correctly to APIs when adding custom rpc support
+      const requests = await Promise.all(
+        uniqueAddresses.map((address) =>
+          fetch(
+            `https://api${appSettings.peek().relayEndpoint === MEV_RELAY_GOERLI ? '-goerli' : ''
+            }.etherscan.io/api?module=contract&action=getabi&address=${address}&apiKey=PSW8C433Q667DVEX5BCRMGNAH9FSGFZ7Q8`,
+          ),
+        ),
+      )
+      const abis = await Promise.all(requests.map((request) => request.json()))
+      const interfaces: { [address: string]: Interface } = abis.reduce((acc, curr: { status: string; result: string }, index) => {
+        if (curr.status === '1') return { ...acc, [`${uniqueAddresses[index]}`]: new utils.Interface(curr.result) }
+        else return acc
+      }, {})
+      const parsed = transactions.value.map((tx) => {
+        if (tx.transaction.to && tx.transaction.data && tx.transaction.data !== '0x' && tx.transaction.data.length > 0) {
+          const decoded = formatTransactionDescription(
+            interfaces[tx.transaction.to].parseTransaction({ ...tx.transaction, data: tx.transaction.data.toString() }),
+          )
+          return { ...tx, decoded }
+        }
+        return tx
+      })
+      parsedTransactions.value = parsed
+    } catch (error) {
+      console.log('parseTransactionsCb Error:', error)
+      parsedTransactions.value = transactions.peek()
+    }
+  }
+  useCallback(parseTransactionsCb, [interceptorPayload.value])
+  parseTransactionsCb()
 
-	return (
-		<>
-			<h2 className='font-bold text-2xl'>Your Transactions</h2>
-			<TransactionList {...{ transactions, fundingTx: interceptorPayload.peek()?.containsFundingTx ?? false }} />
-		</>
-	)
+  return (
+    <>
+      <h2 className='font-bold text-2xl'>Your Transactions</h2>
+      <TransactionList {...{ parsedTransactions, fundingTx: interceptorPayload.peek()?.containsFundingTx ?? false }} />
+    </>
+  )
 }
```

Comment:
Indentation style is wrong for this file now.  `.editorconfig` appears to correctly match other files in the repository.  If you aren't already, I recommend adding an EditorConfig support plugin to your editor so it will automatically do the right thing instead of doing the wrong thing.